File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.min = inf
        self.prev = None
        
    def dfs(self, root: Optional[TreeNode]) -> int:
        if not root:
            return
        self.dfs(root.left)
        if self.prev != None:
            self.min = min(self.min, root.val-self.prev)
        self.prev = root.val
        self.dfs(root.right)
        return self.min
    
    def inorder(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        return [*self.inorder(root.left), root.val, *self.inorder(root.right)]
    
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        res, arr = inf, self.inorder(root)
        for i in range(0, len(arr)-1):
            res = min(res, arr[i+1]-arr[i])
        return res
        
        #
        
        return self.dfs(root)

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function getMinimumDifference(root: TreeNode | null): number {
    let res = Infinity, prev = null
    dfs(root)
    return res
    
    function dfs(root: TreeNode | null) {
        if (!root) return
        dfs(root.left)
        if (prev !== null) res = Math.min(res, root.val-prev)
        prev = root.val
        dfs(root.right)
    }
};